internal/urlparser.go

```diff
@@ -0,0 +1,107 @@
+package internal
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+func ParseEngineURL(engine string) (*url.URL) {
```

Comment:
Seems like we always pass `viper.GetString("engine")` to this method, so maybe the method should not take any params and just be a getter?